### PR TITLE
Sm/custom-label-merging-bug

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
major rewrite of nonDecomposed logic (smaller, simpler)
NUT updates over here: https://github.com/salesforcecli/plugin-source/pull/389

*side effects*
- rename man UT to our normal naming convention (instead of relying on index.js inside test folders)
- add DI of `tree` into ConvertContext.finalize to better support UT (similar to that used elsewhere in sDR)

QA Notes:
worth retesting with STL/89 steps that found the bug

### What issues does this PR fix or reference?
@W-10322335@

### Functionality Before
Willie found the issue here https://github.com/forcedotcom/source-tracking/pull/89#issuecomment-1011176456

when retrieving a single label to an existing file, SDR would remove all the other labels in that file

### Functionality After
proper merging of customlabels

### CODE REVIEW

File: src/convert/convertContext.ts [hunk 1]
```diff
@@ -182,175 +175,161 @@ type ChildIndex = {
...
-      if (!parent[groupName]) {
-        parent[groupName] = [];
-      }
+    this.mergeMap = new Map(result);
+  }
 
-      const group = normalizeToArray(parent[groupName]) as JsonArray;
-      group.push(child);
-    }
+  // private async getMappedChildren(component: SourceComponent):
```
Comment (WillieRuemmele) [inline src/convert/convertContext.ts:307]: NIT: remove

File: test/convert/convertContext.test.ts [hunk 2]
```diff
@@ -283,119 +291,134 @@ describe('Convert Transaction Constructs', () => {
...
+            [nonDecomposed.CHILD_1_NAME, nonDecomposed.CHILD_1_XML],
+            [nonDecomposed.CHILD_2_NAME, nonDecomposed.CHILD_2_XML],
+          ]);
+          state.exampleComponent = component;
         });
 
-        const result = await context.nonDecomposition.finalize(nonDecomposed.DEFAULT_DIR);
-
+        const result = await context.nonDecomposition.finalize(nonDecomposed.DEFAULT_DIR, TREE);
+        // eslint-disable-next-line no-console
```
Comment (WillieRuemmele) [inline test/convert/convertContext.test.ts:302]: no longer needed?

File: test/convert/transformers/nonDecomposedMetadataTransformer.test.ts [hunk 3]
```diff
@@ -29,12 +29,16 @@ describe('NonDecomposedMetadataTransformer', () => {
...
+      const expected = JSON.parse(
+        JSON.stringify({
+          [component.fullName]: {
+            component,
+            children: new ComponentSet([child1, child2]),
+          },
+        })
+      );
+      expect(JSON.parse(JSON.stringify(context.recomposition.state))).to.deep.equal(expected);
+      // expect(context.recomposition.state).to.deep.equal(expected);
```
Comment (WillieRuemmele) [inline test/convert/transformers/nonDecomposedMetadataTransformer.test.ts:41]: is this `expect` needed, or can it be removed?

**Review by WillieRuemmele — COMMENTED** [review]
a few small leftovers. It all looks good to me. Your new comments help explain it better than before

**Review by WillieRuemmele — APPROVED** [review]

**Review by shetzel — APPROVED** [review]

**Pull request merged**